src/UI/EditorDiffMarkers.re

```diff
@@ -0,0 +1,129 @@
+open Oni_Core;
+open Utility;
+
+open Revery.Draw;
+
+[@deriving show({with_path: false})]
+type t = array(marker)
+
+and marker =
+  | Modified
+  | Added
+  | DeletedBefore
+  | DeletedAfter
+  | Unmodified;
+
+let generate = buffer =>
```

Comment:
It would be nice to have a benchmark for this algorithm, so we could get a sense of the relative performance (especially for very large files - like 10k + lines).

Reply:
It seems like this algorithm will always need to traverse the full set of buffer lines - which worries me in terms of render performance

Reply:
Relative to what? I can certainly add a benchmark case for it, but not sure what to compare it to initially.

Reply:
Sorry, relative was the wrong choice of words! I really want to get a sense of the performance 'limits' - in other words, at what size of buffer do we need to start worrying about a non-negligible impact on rendering perf? 

The cases that are most interesting to me are the extreme 'large file' cases, ie:
- 10k line file, diffed against another 10k line file
- 100k line file, diffed against another 100k line file

It'd also be interesting to see cases where there is a large disparity:
- 1 line file diffed against a 10k line file, and vice versa

Really I just want to get a sense of the timing performance (ie, how many ms to diff a 100k file)?

Reply:
Alright, will do!